### PR TITLE
[Windows] Make D3D12 loading dynamic to support pre-Windows 10 versions.

### LINKED ISSUE #88466: [Win7] Godot 4.3dev no longer launches (missing d3d12.dll)
### Tested versions

Reproducible in 4.3 dev3 (I can't find/test older versions of 4.3dev, nor build on this laptop)
Works in 4.2.1-stable
Works in 4.2.2rc1

### System information

Godot v4.3dev3 - Windows 6.1.7601 - Vulkan (Mobile) - dedicated GeForce GTX 970M () - Intel(R) Core(TM) i7-4720HQ CPU @ 2.60GHz (8 Threads)

### Issue description

Project list and editor and export templates all fail to launch on Windows 7 due to `The program can't start because d3d12.dll is missing from your computer.` (etc, you know the dialog).

It appears there is no support for d3d12.dll on Windows 7 (re:. [this support answer](https://answers.microsoft.com/en-us/windows/forum/all/missing-dll-d3d12dll/a11cca24-47f6-462d-b5a4-e7e28ee1bd18)).

I assume this break in support isn't intentional. I hope dynamic loading or weak linking can make this work (if I didn't miss somewhere any existing compatibility/setup instructions) for the benefit of those of us still entertaining this OS :)

### Steps to reproduce

Try to launch any of the Godot editor or export template executables on Windows 7.

### Minimal reproduction project (MRP)

n/a

### PR DESCRIPTION
**Pull request by bruvzg**
Fixes https://github.com/godotengine/godot/issues/88466, supersede https://github.com/godotengine/godot/pull/88486

Also include checks and fixes from https://github.com/godotengine/godot/pull/88486

### CODE REVIEW

**Comment by bruvzg** [comment]
Tested build with D3D12 enabled on Windows 7, it is loading (D3D12 obviously is not working):

capture

**Comment by RandomShaper** [comment]
I would like to take a step back for a moment and reassess if dynamic loading is worth it, considering the amount of patches and compexity it's adding. What's the advantage to it?

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

File: drivers/d3d12/rendering_context_driver_d3d12.cpp [hunk 1]
```diff
@@ -109,26 +120,35 @@ Error RenderingContextDriverD3D12::_init_device_factory() {
...
 		sdk_config->Release();
 	}
-
 	return OK;
 }
 
 Error RenderingContextDriverD3D12::_initialize_debug_layers() {
 	ComPtr<ID3D12Debug> debug_controller;
 	HRESULT res;
+
```
Comment (verypleasentusername) [inline drivers/d3d12/rendering_context_driver_d3d12.cpp:129]: ?

**Pull request merged**